packages/editor/src/components/provider/use-block-editor-settings.js

```diff
@@ -61,11 +63,51 @@ function useBlockEditorSettings( settings, hasTemplate ) {
 			),
 			hasResolvedLocalSiteData: hasFinishedResolvingSiteData,
 			baseUrl: siteData?.url || '',
+			userCanCreatePages: select( coreStore ).canUser(
+				'create',
+				'pages'
+			),
+			userCanCreatePosts: select( coreStore ).canUser(
+				'create',
+				'posts'
+			),
 		};
 	}, [] );
 
 	const { undo } = useDispatch( editorStore );
 
+	const { saveEntityRecord } = useDispatch( coreStore );
+
+	/**
+	 * Creates a Post entity.
+	 * This is utilised by the Link UI to allow for on-the-fly creation of Posts/Pages.
+	 *
+	 * @param {string} postType the post type of the "post" to be created.
+	 * @param {Object} options  parameters for the post being created. These mirror those used on 3rd param of saveEntityRecord.
+	 * @return {Object} the post type object that was created.
+	 */
+	const createEntity = ( postType, options ) => {
+		const postTypeWhitelist = [ 'post', 'page' ];
```

Comment:
Currently I've locked this down to posts and pages but I've left the _code_ below more agnostic so it's easier to extend in the future.

Reply:
I've locked this down to posts only now.